crates/matrix-sdk/src/encryption/backups/mod.rs

```diff
@@ -423,6 +423,39 @@ impl Backups {
         Ok(())
     }
 
+    /// Deletes the current backup version from the server
+    pub async fn delete_backup(&self) -> Result<(), Error> {
+        let _guard = self.client.locks().backup_modify_lock.lock().await;
+
+        self.set_state(BackupState::Disabling);
+
+        // Create a future so we can catch errors and go back to the `Unknown` state.
+        let future = async {
+            let response = self.get_current_version().await?;
+
+            if let Some(response) = response {
+                self.delete_backup_from_server(response.version).await?;
+            }
+
+            let olm_machine = self.client.olm_machine().await;
+            let olm_machine = olm_machine.as_ref().ok_or(Error::NoOlmMachine)?;
+
+            olm_machine.backup_machine().disable_backup().await?;
+
+            self.set_state(BackupState::Unknown);
```

Comment:
Both the OK and error case go into the `Unknown` state, so you can just have one `set_state()` in the main part of the function, just before you return the `result`.